detox/src/devices/allocation/drivers/android/emulator/launchEmulatorProcess.js

```diff
@@ -46,7 +47,23 @@ function launchEmulatorProcess(emulatorName, emulatorExec, emulatorLaunchCommand
 
   log = log.child({ child_pid: childProcessPromise.childProcess.pid });
 
-  return childProcessPromise.then(() => true).catch((err) => {
+  // On Android SDK 31 `childProcessPromise` never resolves when Emulator boots,
+  // so we have to continuosly scan for the process output until we detect that the emulator has booted.
+  const monitorOutputWithTimeout = async (timeoutSeconds) => {
+    for (let i = 0; i < timeoutSeconds; i++) {
+      const stdoutContent = fs.readFileSync(tempLog, 'utf8');
+      if (stdoutContent.includes('boot completed')) {
+        return;
+      }
+
+      await sleep(1000);
+    }
+    throw new Error('Could not boot emulator');
+  };
+  const monitorOutputWithTimeoutPromise = monitorOutputWithTimeout(180);
```

Comment:
@capezzbr Thanks for the contribution. I think 3 minutes may be too long a timeout, and 60 seconds could be enough since if `boot completed` is not found it implies that the device is performing quick boot (which should be 6-7 seconds)